x-pack/filebeat/otel/exporter_factory.go

```diff
@@ -0,0 +1,205 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+//
+// The MetricsExporterFactory reads environment variables and creates the metrics exporter requested
+// by the environment.
+//
+// To export OTEL metrics to an OTLP/gRPC endpoint set these environment variables:
+//   - OTEL_EXPORTER_OTLP_ENDPOINT: Required. The OTLP endpoint URL.
+//   - OTEL_EXPORTER_OTLP_HEADERS: Required if endpoint is authenticated.
+//
+// To export using the httt/protobuf protocol add this environment variable with the above environment variables:
+// - OTEL_EXPORTER_OTLP_METRICS_PROTOCOL="http/protobuf"
+//
+// To override other environment variables and print a JSON representation of the metrics to console set:
+// - OTEL_METRICS_EXPORTER=console
+//
+// To override other environment variables and disable OTEL metrics collection set:
+// - OTEL_METRICS_EXPORTER=none
+package otel
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"strings"
+	"sync"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+// init() Creates a singleton MetricsExporterFactory that can be used to create
+// global exporters.
+func init() {
+	exporterFactory = NewMetricsExporterFactory(GetDefaultMetricExporterOptions())
+}
```

Comment:
I'd rather use a sync.Once in GetGlobalMetricsExporterFactory instead of an init function. We had lots of issues bitting us because of init functions.